src/utils/link-channel.js

```diff
@@ -0,0 +1,141 @@
+import { generatePublicKeyAndEncryptedObject, generateKeys, decryptObject } from "./crypto";
+
+export default class LinkChannel {
+  constructor(store) {
+    this.store = store;
+  }
+
+  setSocket = socket => {
+    this.socket = socket;
+  };
+
+  // Returns a promise that, when resolved, will forward an object with three keys:
+  //
+  // code: The code that was made available to use for link.
+  //
+  // cancel: A function that the caller can call to cancel the use of the code.
+  //
+  // onFinished: A promise that, when resolved, indicates the code is no longer usable,
+  // because it was either successfully used by the remote device or it has expired
+  // ("used" or "expired" is passed to the callback.)
+  generateCode = () => {
+    return new Promise(resolve => {
+      const onFinished = new Promise(finished => {
+        const step = () => {
+          const code = Math.floor(Math.random() * 9999)
+            .toString()
+            .padStart(4, "0");
+
+          // Only respond to one link_request in this channel.
+          let readyToSend = false;
+          let leftChannel = false;
+
+          const channel = this.socket.channel(`link:${code}`, { timeout: 10000 });
+
+          const leave = () => {
+            if (!leftChannel) channel.leave();
+            leftChannel = true;
+          };
+
+          const cancel = () => leave();
```

Comment:
just `const cancel = leave;`

Reply:
Haha I wrote this code unsure if your suggestion was idiomatic or not, and thought "Brian will probably say so in the PR" :)